index.php

```diff
@@ -69,26 +74,28 @@ function Modcast()
 	}
 
 	$json->timestamp = time();
+	$json->items = count($channel->videos);
 	$json->data = podcastRSSFeed;
 	return $json;
 }
 
-// only sends request if stream expired
+// fetchs rss on every request and if audio uris expired
 // gets data from cache if unexpired
 // serves podcast rss feed
+// checks new videos if available
 $expiresInSeconds = 21540;
 if (file_exists("cache")) {
 	$cache = file_get_contents("cache");
 	$cache = json_decode($cache);
-	if (time() - $cache->timestamp > $expiresInSeconds) {
-		$data = Modcast()->data;
+	if (time() - $cache->timestamp > $expiresInSeconds || $cache->items < count(getFeed()->videos)) {
```

Comment:
Unfortunately this check won't work because YouTube RSS' only gives 15 items of a list. So when it reached 15, it will always return 15. Also this isn't a good approach because maybe creator of playlist removed an item and added a new one between our checks. Right? I think this all newly adapted changes should be decided before.

Reply:
You're right, I didn't notice YouTube RSS returns 15 items. No information in RSS about last playlist update time maybe checking difference with video titles or ids will be better